File: fib.php
<?php

function fibonacci($n) {
    if ($n <= 1) return gmp_init($n);
    $a = gmp_init(0);
    $b = gmp_init(1);
    for ($i = 2; $i <= $n; $i++) {
        $temp = gmp_add($a, $b);
        $a = $b;
        $b = $temp;
    }
    return $b;
}

$startTime = microtime(true);
$result = fibonacci(100);
$endTime = microtime(true);

$timeTaken = $endTime - $startTime;
$memoryUsed = memory_get_peak_usage(true) / 1024;

echo "Language: PHP\n";
printf("Time: %.6f seconds\n", $timeTaken);
printf("Memory: %d KB\n", $memoryUsed);
echo "Result: " . gmp_strval($result) . "\n";
?>

File: fib.c
#include <stdio.h>
#include <time.h>
#include <sys/resource.h>

long long fibonacci(int n) {
    if (n <= 1) return n;
    long long a = 0, b = 1, temp;
    for (int i = 2; i <= n; i++) {
        temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

int main() {
    struct timespec start, end;
    struct rusage usage;
    
    clock_gettime(CLOCK_MONOTONIC, &start);
    long long result = fibonacci(50);
    clock_gettime(CLOCK_MONOTONIC, &end);
    
    double time_taken = (end.tv_sec - start.tv_sec) + 
                        (end.tv_nsec - start.tv_nsec) / 1e9;
    
    getrusage(RUSAGE_SELF, &usage);
    long memory_kb = usage.ru_maxrss;
    
    printf("Language: C\n");
    printf("Time: %.6f seconds\n", time_taken);
    printf("Memory: %ld KB\n", memory_kb);
    printf("Result: %lld\n", result);
    
    return 0;
}
